tests/test_poly.py

```diff
@@ -0,0 +1,287 @@
+    pos = [ (numpy.random.random()*X, numpy.random.random()*Y)
+            for i in range(nstars) ]
+
+    for i in xrange(nparam):
+        param = numpy.random.random(size=nstars)
+        p0 = interp.initialGuess(pos, param, i)
+        mu = param.mean()
+        assert numpy.isclose(p0[0,0],mu)
+        assert (p0[0,1:]==0.0).all()
+        assert (p0[1,0]==0.0).all()
+        assert (p0[1:,1:]==0.0).all()
+        assert numpy.allclose(interp.interpolationModel(pos, p0), mu)
+
+
+
+def poly_load_save_sub(type1, type2):
+    #Test that we can serialize and deserialize a polynomial 
+    #interpolator correctly.  Copying all this stuff from above:
+
+    numpy.random.seed(12434)
+    nparam = 3
+    nstars=50   
+    #Use three different sizes to test everything
+    orders = [1,2,3]
+    interp = piff.Polynomial(orders, poly_type=type1)
+    X = 10.0 #size of the field
+    Y = 10.0
+
+    pos = [ (numpy.random.random()*X, numpy.random.random()*Y)
+            for i in range(nstars) ]
+
+    #Let's make a function that is linear just as a function of one parameter
+    #These are the linear fit parameters for each parameter in turn
+    m1 = numpy.random.uniform(size=nparam)
+    m2 = numpy.random.uniform(size=nparam)
+    q1 = numpy.random.uniform(size=nparam)
+    c = numpy.random.uniform(size=nparam)
+
+    def f(pos):
+        u = pos[0]
+        v = pos[1]
+        r = q1*u*v+ m1*u+m2*v+c
+        return r
+
+    #Simulate the vectors under this model
+    vectors = [f(p) for p in pos]
+
+    #Fit them!
+    interp.solve(pos, vectors)
+
+    #We should overwrite the order parameter when we load in
+    interp2 = piff.Polynomial([0], poly_type=type2)
+
+
+    import tempfile
+    import os
+    import fitsio
+    extname = "interp"
+    dirname = tempfile.mkdtemp()
+    filename=os.path.join(dirname,'poly_test_file.fits')
+    with fitsio.FITS(filename,'rw',clobber=False) as f:
+        interp.writeSolution(f, extname=extname)
+    f2 = fitsio.FITS(filename, "r")
+    interp2.readSolution(f2, extname=extname)
+    os.remove(filename)
+    os.rmdir(dirname)
+
+
+    #The type and other parameters should now have been overwritten and updated
+    assert interp2.poly_type == interp.poly_type
+    assert interp2.orders==interp.orders
+    assert interp2.nvariables==interp.nvariables
+    assert interp2.indices==interp.indices
+
+    #Check that the old and new interpolators generate the same
+    #value
+    for i in xrange(30):
+        p=(numpy.random.random()*X, numpy.random.random()*Y)
+        assert numpy.allclose(interp.interpolate(p),interp2.interpolate(p))
+
+def test_poly_load_save():
+    for poly_type in PolynomialsTypes:
+        poly_load_save_sub(poly_type,poly_type)
+
+def test_poly_load_err():
+    from nose.tools import assert_raises
+    for poly_type1 in PolynomialsTypes[:]:
+        for poly_type2 in PolynomialsTypes[:]:
+            if poly_type1!=poly_type2:
+                poly_load_save_sub(poly_type1,poly_type2)
```

Comment:
Should put a bit at the bottom that lets you run this from the command line:

if __name__ == '__main__':
    test_poly_indexing()
    test_poly_mean()
    test_poly_linear()
    test_poly_quadratic()
    ...
```